### DISCUSSION TITLE
Grid_SizeChanged getting called repetitively

### DISCUSSION THREAD

**Discussion by gitamo626** [abbaye/WPFHexEditorControl#132] [Q&A]
Hi there - 

I've got the following xaml:
`<Grid>`
`        <Grid.RowDefinitions>`
`            <RowDefinition Height="Auto"/>`
`            <RowDefinition Height="Auto"/>`
`            <RowDefinition  />`
`            <RowDefinition Height="Auto" />`
`            <RowDefinition Height="Auto" />`
`        </Grid.RowDefinitions>`
`        <Menu Grid.Row="0">`
`            <Menu.CommandBindings>`
`                <CommandBinding Command="{x:Static ApplicationCommands.Close}"`
`                                Executed="CommandBinding_Executed" />`
`            </Menu.CommandBindings>`
`            <MenuItem x:Name="mnuFile" Header="File">`
`                <MenuItem Header="Exit" Command="{x:Static ApplicationCommands.Close}"  />`
`            </MenuItem>`
`        </Menu>`
`        <Grid Grid.Row="1" Margin="3">`
`            <Grid.ColumnDefinitions>`
`                <ColumnDefinition Width="Auto"/>`
`                <ColumnDefinition/>`
`                <ColumnDefinition Width="Auto"/>`
`            </Grid.ColumnDefinitions>`
`            <TextBlock Text="File to Inspect" Padding="2,0" />`
`            <TextBox Grid.Column="1" x:Name="txtInputFile"  />`
`            <Button x:Name="btnBrowse" Grid.Column="2" Content="Browse" Padding="2,0" Margin="2,0" Click="btnBrowse_Click"/>`
`        </Grid>`
`        <ListView x:Name="lvPE" Grid.Row="2">`
`            <ListView.View>`
`                <GridView>`
`                    <GridViewColumn Header="Field Name" DisplayMemberBinding="{Binding Key}" />`
`                    <GridViewColumn Header="Value" DisplayMemberBinding="{Binding Value}" />`
`                </GridView>`
`            </ListView.View>`
`        </ListView>`
`        <GridSplitter HorizontalAlignment="Stretch" Height="5" Background="LightGray" BorderThickness="1" BorderBrush="Gray" Grid.Row="3" />`
`        <hex:HexEditor x:Name="hexEditor" Grid.Row="4" Height="300" FontFamily="Consolas" FontSize="14" AllowAutoHighLightSelectionByte="False" />`

`    </Grid>`

If I set MinHeight (going from Height to MinHeight) on the hex editor control, I run into issues even in the design mode. The UI starts locking up. At some point, a message pops up "XAML designer is busy". On that little popup dialog, if I click the Cancel button, the UI will unfreeze. However, that's not the end of it all. At runtime, I've stepped through the code (somewhat) and noticed that Grid_SizeChanged event handler is getting invoked repetitively which in turns results into multiple calls to RefreshView method. The crescendo of these calls will lock up the VS (2022, Version 17.4.3) and it needs to be force terminated.

I've tried to fix the issue on my own but am not familiar with the design of the control. Anyone got an idea how to go about fixing this? 
![xaml-designer-busy](https://user-images.githubusercontent.com/106823221/208955063-bd3621cc-d081-4938-972f-6289c601ceab.png)

**Comment by abbaye** [comment]
Corrected
